pennylane/templates/embeddings.py

```diff
@@ -93,28 +93,41 @@ def AmplitudeEmbedding(features, wires, pad=False, normalize=False):
         normalize (Boolean): controls the activation of automatic normalization
     """
 
-    if not isinstance(wires, Iterable):
-        raise ValueError("Wires needs to be a list of wires that the embedding uses; got {}.".format(wires))
+    if isinstance(wires, int):
+        wires = [wires]
+
+    features = np.array(features)
 
     n_features = len(features)
     n_amplitudes = 2**len(wires)
 
     if n_amplitudes < n_features:
-        raise ValueError("AmplitudeEmbedding requires the size of feature vector to be smaller than or equal to 2**len(wires), which is {}; "
+        raise ValueError("AmplitudeEmbedding requires the size of feature vector to be "
+                         "smaller than or equal to 2**len(wires), which is {}; "
                          "got {}.".format(n_amplitudes, n_features))
 
     if pad and n_amplitudes >= n_features:
         features = np.pad(features, (0, n_amplitudes-n_features), 'constant')
 
-    if  not pad and n_amplitudes != n_features:
-        raise ValueError("AmplitudeEmbedding with no padding requires a feature vector of size 2**len(wires), which is {}; "
-                         "got {}.".format(n_amplitudes, n_features))
+    if not pad and n_amplitudes != n_features:
+        raise ValueError("AmplitudeEmbedding must get a feature vector of size 2**len(wires), "
+                         "which is {}; got {}. Use ``pad=True`` to automatically pad the "
+                         "features with zeros.".format(n_amplitudes, n_features))
 
-    if normalize and np.linalg.norm(features, 2) != 1:
-        features = features * (1/np.linalg.norm(features, 2))
+    # Get normalization
+    norm = 0
+    for f in features:
```

Comment:
I suppose this will break differentiability for any feature that is encoded as a tensorflow or pytorch tensor?

Reply:
Ah, shoot. You are right. But how else would we do it? Check for the type (which is very restrictive)? Note that this is a general problem in templates, how can we interact with inputs without assuming too much.

Reply:
A more drastic option is to remove normalization altogether, and leave it up to the plugins to check. Previously, the problem was that _nothing_ was checking normalization, but this has been slowly added to all plugins (#346 adds this to default.qubit, not yet merged).

This way the user still gets a useful error message, and the checking occurs at a point where autodiff 'doesn't matter' per se.

Reply:
Some thoughts on how to deal with this:

- we have a dictionary which maps different arithmetic operations to the equivalent in the relevant library (e.g., https://www.tensorflow.org/api_docs/python/tf/math/l2_normalize). This works, but is clunky (and requires checking if libraries are installed via `try/except`.

- since all those libraries should overload the basic arithmetic operations, we could do it implicitly, e.g., `sum(array ** 2)`. If `array` is a list (or numpy array) of `Tensor`s, this should work. Otherwise if array itself is a `Tensor`, you would have to manually do the sum: `sum([b[idx] for idx in range(b.shape[0])])`.

Reply:
> I suppose this will break differentiability for any feature that is encoded as a tensorflow or pytorch tensor?

Actually just realised that nothing should be a Tensor at this point in the code. The QNode converts from Tensor -> NumPy array -> Variable (if differentiable).

Reply:
Then we just need to make sure that `+` and `**` are overloaded for pennylane `Variable`s

Reply:
Actually, the first thing to do is to write integration tests for the embeddings, to check if the interfaces pass through. 
I only realize this now, the templates were supposed to have had them all along.

Reply:
I tested the torch and tf interface, and as @josh146 said everything works fine with the new manual normalization, the operators must be already overloaded.